spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -1586,6 +1586,18 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
       case StringTrimBoth(srcStr, trimStr, _) =>
         trim(expr, srcStr, trimStr, inputs, binding, "btrim")
 
+      case StringLPad(srcStr, size, chars) =>
+        val arg0 = exprToProtoInternal(srcStr, inputs, binding)
+        val arg1 = exprToProtoInternal(Cast(size, LongType), inputs, binding)
+        val arg2 = exprToProtoInternal(chars, inputs, binding)
+        scalarExprToProto("lpad", arg0, arg1, arg2)
+
+      case StringRPad(srcStr, size, chars) =>
+        val arg0 = exprToProtoInternal(srcStr, inputs, binding)
+        val arg1 = exprToProtoInternal(Cast(size, LongType), inputs, binding)
+        val arg2 = exprToProtoInternal(chars, inputs, binding)
+        scalarExprToProto("rpad", arg0, arg1, arg2)
```

Comment:
I remember that DataFusion rpad truncates the string if it is already longer than the `size`. Could you add a test if Spark behaves the same?

Reply:
- Spark's `rpad` does also truncate. However, DataFusion's `rpad` is not compatible with Spark for the unicode inputs you mentioned, so I will need to implement a custom version :disappointed:.

Reply:
https://github.com/apache/datafusion-comet/blob/main/native/spark-expr/src/static_invoke/char_varchar_utils/read_side_padding.rs#L60
I remember DataFusion uses graphemes vs Spark uses chars. Perhaps best to check postgres if it uses chars. If so, we may fix in DataFusion?
chars has better performance as well

Reply:
Yes, DataFusion's `rpad` implementation works on graphemes. I am not 100% sure but I think Postgres's version works on unicode code points.